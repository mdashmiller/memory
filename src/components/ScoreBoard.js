import React from 'react'
import styled from 'styled-components'
import emojisArr from '../assets/images-arr'
import PropTypes from 'prop-types'

const EmptyCircle = styled.div`
	width: 22px;
	height: 22px;
	margin: 10px 0 0;
	display: inline-block;
	border: 1px solid #bbc4ef;
	border-radius: 13px;
`

const LilSmile = styled.img`
	opacity: 1;
	width: 24px;
	height: 24px;
	display: inline-block;
	margin: 10px 0 0;
`

const ScoreBoard = ({ posMatches }) =>
	// empty circles fill-in with smiley faces
	// for each match the user makes
	<div style={{ textAlign: 'center' }}>
		{posMatches.length > 1
			? <LilSmile src={emojisArr[39]} alt="smiley face" className="smile"></LilSmile>
			: <EmptyCircle className="empty"></EmptyCircle>
		}
		{posMatches.length > 3
			? <LilSmile src={emojisArr[39]} alt="smiley face" className="smile"></LilSmile>
			: <EmptyCircle className="empty"></EmptyCircle>
		}
		{posMatches.length > 5
			? <LilSmile src={emojisArr[39]} alt="smiley face" className="smile"></LilSmile>
			: <EmptyCircle className="empty"></EmptyCircle>
		}
		{posMatches.length > 7
			? <LilSmile src={emojisArr[39]} alt="smiley face" className="smile"></LilSmile>
			: <EmptyCircle className="empty"></EmptyCircle>
		}
		{posMatches.length > 9
			? <LilSmile src={emojisArr[39]} alt="smiley face" className="smile"></LilSmile>
			: <EmptyCircle className="empty"></EmptyCircle>
		}
		{posMatches.length > 11
			? <LilSmile src={emojisArr[39]} alt="smiley face" className="smile"></LilSmile>
			: <EmptyCircle className="empty"></EmptyCircle>
		}
	</div>

ScoreBoard.propTypes = {
	posMatches: PropTypes.arrayOf(PropTypes.number).isRequired
}

export default ScoreBoard
