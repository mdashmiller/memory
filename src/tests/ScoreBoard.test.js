import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import ScoreBoard from '../components/ScoreBoard'

Enzyme.configure({ adapter: new Adapter() })

describe('<ScoreBoard />', () => {

	const posMatches = []
	let matches

	it('renders without crashing', () => {
		const div = document.createElement('div')
		ReactDOM.render(<ScoreBoard posMatches={posMatches} />, div)
		ReactDOM.unmountComponentAtNode(div)
	})

	test('has a valid snapshot', () => {
		const component = renderer.create(
			<ScoreBoard posMatches={posMatches} />
		)
		let tree = component.toJSON()
		expect(tree).toMatchSnapshot()
	})

	it('renders empty circles at all positions', () => {
		const wrapper = shallow(<ScoreBoard posMatches={posMatches} />)

		expect(wrapper.find('.empty').length).toBe(6)
	})

	it('renders 1 smile with 1 position match', () => {
		matches = [1, 2]
		posMatches.push(...matches)
		const wrapper = shallow(<ScoreBoard posMatches={posMatches} />)

		expect(wrapper.find('.smile').length).toBe(1)
	})

	it('renders 2 smiles with 2 position matches', () => {
		matches = [3, 4]
		posMatches.push(...matches)
		const wrapper = shallow(<ScoreBoard posMatches={posMatches} />)

		expect(wrapper.find('.smile').length).toBe(2)
	})

	it('renders 3 smiles with 3 position matches', () => {
		matches = [5, 6]
		posMatches.push(...matches)
		const wrapper = shallow(<ScoreBoard posMatches={posMatches} />)

		expect(wrapper.find('.smile').length).toBe(3)
	})

	it('renders 4 smiles with 4 position matches', () => {
		matches = [7, 8]
		posMatches.push(...matches)
		const wrapper = shallow(<ScoreBoard posMatches={posMatches} />)

		expect(wrapper.find('.smile').length).toBe(4)
	})

	it('renders 5 smiles with 5 position matches', () => {
		matches = [9, 10]
		posMatches.push(...matches)
		const wrapper = shallow(<ScoreBoard posMatches={posMatches} />)

		expect(wrapper.find('.smile').length).toBe(5)
	})

	it('renders 6 smiles with 6 position matches', () => {
		matches = [11, 12]
		posMatches.push(...matches)
		const wrapper = shallow(<ScoreBoard posMatches={posMatches} />)
		
		expect(wrapper.find('.smile').length).toBe(6)
	})

})
