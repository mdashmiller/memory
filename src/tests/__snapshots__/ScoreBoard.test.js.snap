// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScoreBoard /> has a valid snapshot 1`] = `
<div
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <div
    className="empty sc-bdVaJa jyRWkp"
  />
  <div
    className="empty sc-bdVaJa jyRWkp"
  />
  <div
    className="empty sc-bdVaJa jyRWkp"
  />
  <div
    className="empty sc-bdVaJa jyRWkp"
  />
  <div
    className="empty sc-bdVaJa jyRWkp"
  />
  <div
    className="empty sc-bdVaJa jyRWkp"
  />
</div>
`;
